src/components/MyObservations/hooks/useUploadObservations.ts

```diff
@@ -0,0 +1,237 @@
+import { useNavigation, useRoute } from "@react-navigation/native";
+import { deactivateKeepAwake } from "@sayem314/react-native-keep-awake";
+import { RealmContext } from "providers/contexts";
+import {
+  useCallback, useEffect, useState
+} from "react";
+import { EventRegister } from "react-native-event-listeners";
+import Observation from "realmModels/Observation";
+import {
+  INCREMENT_SINGLE_UPLOAD_PROGRESS
+} from "sharedHelpers/emitUploadProgress";
+import uploadObservation, { handleUploadError } from "sharedHelpers/uploadObservation";
+import {
+  useTranslation
+} from "sharedHooks";
+import useStore from "stores/useStore";
+
+const { useRealm } = RealmContext;
+
+export default useUploadObservations = (
+  toggleLoginSheet,
+  showInternetErrorAlert,
```

Comment:
FWIW, I don't think this hook needs to have any awareness of UI logic like this. If there's no internet connection, it should either throw an error or include an error in its return value, and the component using this hook can decide what do do with that in the UI. Same goes for not having a logged in user.